spec/requests/v1/pacbio/libraries_spec.rb

```diff
@@ -71,184 +89,225 @@
+          expect(response).to have_http_status(:unprocessable_entity)
         end
 
-      end
+        it 'cannot create a library' do
+          expect { post v1_pacbio_libraries_path, params: body, headers: json_api_headers }.to_not change(Pacbio::Library, :count)
+        end
 
-    end
+        it 'has an error message' do
+          post v1_pacbio_libraries_path, params: body, headers: json_api_headers
+          expect(JSON.parse(response.body)["data"]).to include("errors" => {"volume"=>["can't be blank"]})
+        end
 
+      end
+    end
+ 
     context 'when creating multiple libraries' do
-      let(:request) { create(:pacbio_request) }
-      let(:tag) { create(:pacbio_tag) }
 
       context 'on success' do
-        context 'when the sample does exist' do
-
-          let(:body) do
-            {
-              data: {
-                attributes: {
-                  libraries: [
-                    { volume: 1.11,
-                      concentration: 2.22,
-                      library_kit_barcode: 'LK1234567',
-                      fragment_size: 100,
-                      pacbio_tag_id: tag.id,
-                      pacbio_request_id: request.id
-                    },
-                    { volume: 3.22,
-                      concentration: 4.22,
-                      library_kit_barcode: 'LK1234569',
-                      fragment_size: 200,
-                      pacbio_tag_id: tag.id,
-                      pacbio_request_id: request.id
+
+        let(:body) do
+        {
+          data: {
+            type: 'libraries',
+            attributes: {
+              libraries: [
+                { 
+                    volume: 1.11,
+                    concentration: 2.22,
+                    library_kit_barcode: 'LK1234567',
+                    fragment_size: 100,
+                    relationships: {
+                      requests: {
+                        data: [
+                          { 
+                            type: 'requests', 
+                            id: request.id, 
+                            relationships: {
+                              tag: {
+                                data: {
+                                  type: 'tags',
+                                  id: tag.id
+                                }
+                              }
+                            }
+                          },
+                          { 
+                            type: 'requests', 
+                            id: request.id, 
+                            relationships: {
+                              tag: {
+                                data: {
+                                  type: 'tags',
+                                  id: tag.id
+                                }
+                              }
+                            }
+                          }
+                        ]
+                      }
+                    }
+                },
```

Comment:
Should we add a seperate test for when a library is created with two requests

Reply:
Good spot :) I did have multiple requests wrapped up in another test but I wasn't checking whether those requests existed. Turns out the associations weren't being saved so I have set autosave to true. I am happy to leave it like that. Do you think it needs to be in its own test?

Reply:
If we can just test in the existing test that the multiple requests exist, that is maybe enough? :)

Reply:
Yes. I have added an extra expectation the multiple libraries test which checks that two requests exist.